File: LondonDataServices.IDecide.Manage.Server.Tests.Integration/Models/ConsumerAdoptions/ConsumerAdoption.cs
// ---------------------------------------------------------
// Copyright (c) North East London ICB. All rights reserved.
// ---------------------------------------------------------

using System;

namespace LondonDataServices.IDecide.Manage.Server.Tests.Integration.Models.ConsumerAdoptions
{
    public class ConsumerAdoption
    {
        public Guid Id { get; set; }
        public Guid ConsumerId { get; set; }
        public Guid DecisionId { get; set; }
        public DateTimeOffset AdoptionDate { get; set; }
        public string CreatedBy { get; set; }
        public DateTimeOffset CreatedDate { get; set; }
        public string UpdatedBy { get; set; }
        public DateTimeOffset UpdatedDate { get; set; }
    }
}

File: LondonDataServices.IDecide.Portal.Client/src/models/consumerAdoptions/consumerAdoption.ts
import { Decision } from '../decisions/decision';
import { Consumer } from '../consumers/consumer';

export class ConsumerAdoption {
    public id?: string;
    public consumerId?: string;
    public decisionId?: string;
    public adoptionDate?: Date;
    public createdBy?: string;
    public createdDate?: Date;
    public updatedBy?: string;
    public updatedDate?: Date;
    public consumer?: Consumer;
    public decision?: Decision;

    constructor(adoption?: Partial<ConsumerAdoption>) {
        this.id = adoption?.id || '';
        this.consumerId = adoption?.consumerId || '';
        this.decisionId = adoption?.decisionId || '';
        this.adoptionDate = adoption?.adoptionDate ? new Date(adoption.adoptionDate) : undefined;
        this.createdBy = adoption?.createdBy || '';
        this.createdDate = adoption?.createdDate ? new Date(adoption.createdDate) : undefined;
        this.updatedBy = adoption?.updatedBy || '';
        this.updatedDate = adoption?.updatedDate ? new Date(adoption.updatedDate) : undefined;
        this.consumer = adoption?.consumer;
        this.decision = adoption?.decision;
    }
}